hazelcast/src/main/java/com/hazelcast/query/impl/getters/Extractors.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (c) 2008-2015, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.query.impl.getters;
+
+import com.hazelcast.config.MapAttributeConfig;
+import com.hazelcast.query.extractor.ValueExtractor;
+import com.hazelcast.query.impl.DefaultArgumentParser;
+import com.hazelcast.util.ExceptionUtil;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import static com.hazelcast.query.impl.getters.ExtractorHelper.extractArgumentsFromAttributeName;
+import static com.hazelcast.query.impl.getters.ExtractorHelper.extractAttributeNameNameWithoutArguments;
+import static com.hazelcast.query.impl.getters.ExtractorHelper.instantiateExtractors;
+
+public final class Extractors {
+
+    private static final int MAX_CLASSES_IN_CACHE = 1000;
+    private static final int MAX_GETTERS_PER_CLASS_IN_CACHE = 100;
+    private static final float EVICTION_PERCENTAGE = 0.2f;
+    private static final Extractors EMPTY = new Extractors(Collections.<MapAttributeConfig>emptyList());
+
+    /**
+     * Maps the extractorAttributeName WITHOUT the arguments to a ValueExtractor instance.
+     * The name does not contain the argument since it's not allowed to register an extractor under an attribute name
+     * that contains an argument in square brackets.
+     */
+    private final Map<String, ValueExtractor> extractors;
+    private final EvictableGetterCache getterCache;
+    private final DefaultArgumentParser argumentsParser;
+
+    public Extractors(List<MapAttributeConfig> mapAttributeConfigs) {
+        this.extractors = instantiateExtractors(mapAttributeConfigs);
+        this.getterCache = new EvictableGetterCache(MAX_CLASSES_IN_CACHE, MAX_GETTERS_PER_CLASS_IN_CACHE,
+                EVICTION_PERCENTAGE);
+        this.argumentsParser = new DefaultArgumentParser();
+    }
+
+    public Object extract(Object targetObject, String attributeName) {
+        if (targetObject != null) {
```

Comment:
I believe this could be simplified.
```java
if (targetObject == null) {
  return null;
}
```

It eliminates the second branch. it's a taste thing anyway.